pkg/operator/ceohelpers/bootstrap.go

```diff
@@ -151,26 +151,28 @@ func CheckSafeToScaleCluster(
 }
 
 // IsBootstrapComplete returns true if bootstrap has completed.
-func IsBootstrapComplete(configmapLister corev1listers.ConfigMapLister, staticPodClient v1helpers.StaticPodOperatorClient, etcdClient etcdcli.AllMemberLister) (bool, error) {
+func IsBootstrapComplete(configmapLister corev1listers.ConfigMapLister, staticPodClient v1helpers.StaticPodOperatorClient, etcdClient etcdcli.AllMemberLister, shouldRunRevisionStabilityCheck bool) (bool, error) {
 	// do a cheap check to see if the installer has marked
 	// bootstrapping as done by creating the configmap first.
 	if isBootstrapComplete, err := bootstrap.IsBootstrapComplete(configmapLister); !isBootstrapComplete || err != nil {
 		return isBootstrapComplete, err
 	}
 
 	// now run check to stability of revisions
-	_, status, _, err := staticPodClient.GetStaticPodOperatorState()
-	if err != nil {
-		return false, fmt.Errorf("failed to get static pod operator state: %w", err)
-	}
-	if status.LatestAvailableRevision == 0 {
-		return false, nil
-	}
-	for _, curr := range status.NodeStatuses {
-		if curr.CurrentRevision != status.LatestAvailableRevision {
-			klog.V(4).Infof("bootstrap considered incomplete because revision %d is still in progress", status.LatestAvailableRevision)
+	if shouldRunRevisionStabilityCheck {
+		_, status, _, err := staticPodClient.GetStaticPodOperatorState()
+		if err != nil {
+			return false, fmt.Errorf("failed to get static pod operator state: %w", err)
+		}
+		if status.LatestAvailableRevision == 0 {
```

Comment:
I would start to split out those functions and have one for "IsBootstrapComplete" and one for "IsRevisionStable".

etcdendpointscontroller.go can then just use the IsRevisionStable function alongside the IsBootstrapComplete - the others can stay as they are